content/en/docs/usage/gateway-api.md

```diff
@@ -0,0 +1,248 @@
+
+The Gateway support for Certificate shims must be enabled manually using a flag.
+If you are using Helm:
+
+```sh
+helm upgrade --install cert-manager jetstack/cert-manager
+  --set "extraArgs={--controllers=*\,gateway-shim}"
+```
+
+If you are using the raw cert-manager manifests, you can add the following like
+to the cert-manager controller Deployment `args`:
+
+```sh
+--controllers=*,gateway-shim
+```
+
+Like with an Ingress, the annotations `cert-manager.io/issuer` or
+`cert-manager.io/cluster-issuer` tells cert-manager that this Gateway should be
+looked at to create Certificate shims. For example, the following Gateway should
+trigger the creation of the `example-com-tls` Certificate:
+
+```yaml
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: example
+  annotations:
+    cert-manager.io/issuer: foo
+spec:
+  gatewayClassName: foo
+  listeners:
+    - hostname: example.com
+      port: 443
+      protocol: HTTPS
+      routes:
+        kind: HTTPRoute
+        selector:
+          matchLabels:
+            app: foo
+      tls:
+        mode: Terminate
+        certificateRef:
+          name: example-com-tls
+          kind: Secret
+          group: core
+```
+
+A few moments later, cert-manager should create a Certificate shim:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: example-com-tls
+spec:
+  issuerRef:
+    name: my-issuer
+    kind: Issuer
+    group: cert-manager.io
+  dnsNames:
+    - example.com
+  secretName: example-com-tls
+```
+
+Contrary to Ingresses, you can set the same secret name multiple times. Let us
+imagine that you have these two listeners:
+
+```yaml
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: example
+  annotations:
+    cert-manager.io/issuer: foo
+spec:
+  gatewayClassName: foo
+  listeners:
+    - hostname: example.com
+      port: 443
+      protocol: HTTPS
+      routes:
+        kind: HTTPRoute
+        selector:
+          matchLabels:
+            app: foo
+      tls:
+        mode: Terminate
+        certificateRef:
+          name: example-com-tls
+          kind: Secret
+          group: core
+    - hostname: www.example.com
+      port: 443
+      protocol: HTTPS
+      routes:
+        kind: HTTPRoute
+        selector:
+          matchLabels:
+            app: foo
+      tls:
+        mode: Terminate
+        certificateRef:
+          name: example-com-tls
+          kind: Secret
+          group: core
+```
+
+The Certificate "shim" created by cert-manager will looks like this:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: example-com-tls
+spec:
+  issuerRef:
+    name: my-issuer
+    kind: Issuer
+    group: cert-manager.io
+  dnsNames:
+    - example.com
+    - www.example.com
+  secretName: example-com-tls
+```
+
+The only kind and group supported in `certificateRef` is `Secret` and `core`,
+and both fields are required for the creation of a Certificate shim:
```

Comment:
Suggestion: add somewhere that the 'hostname' field too is required (and maybe also that the DNS name on `Certificate` will be the hostname from the listener- or maybe that's obvious).

Reply:
Good point.

- I added a table that shows all the requirements for a "valid" listener,
- I added a sentence + a comment in the YAML example to clearly show that the hostname field is used for setting the DNS name.